botorch/acquisition/hentropy_search.py

```diff
@@ -0,0 +1,277 @@
+        relative to the values of the objective function.
+
+        Args:
+            loss_function_hyperparameters: hyperparameters for the loss function class.
+        """
+
+        super().__init__()
+        self.dist_weight = dist_weight
+        self.dist_threshold = dist_threshold
+
+    def forward(self, A: Tensor, Y: Tensor) -> Tensor:
+        r"""Evaluate batch loss function on a tensor of actions.
+
+        Args:
+            A: Actor tensor with shape `n_fantasy_at_design_pts x batch_size
+                x num_actions x action_dim`.
+            Y: Fantasized sample with shape `n_fantasy_at_action_pts x
+                n_fantasy_at_design_pts x batch_size x num_actions`.
+
+        Returns:
+            A Tensor of shape `n_fantasy_at_action_pts x batch_size`.
+        """
+
+        Y = Y.sum(dim=-1).mean(dim=0)
+        # >>> n_fantasy_at_design_pts x batch_size
+
+        num_actions = A.shape[-2]
+
+        dist_reward = 0
+        if num_actions >= 2:
+            A = A.contiguous()
+            # >>> n_fantasy_at_design_pts x batch_size x num_actions x action_dim
+
+            A_distance = torch.cdist(A, A, p=1.0)
+            # >>> n_fantasy_at_design_pts x batch_size x num_actions x num_actions
+
+            A_distance_triu = torch.triu(A_distance)
+            # >>> n_fantasy_at_design_pts x batch_size x num_actions x num_actions
+
+            A_distance_triu[A_distance_triu > self.dist_threshold] = self.dist_threshold
+            # >>> n_fantasy_at_design_pts x batch_size x num_actions x num_actions
+
+            denominator = num_actions * (num_actions - 1) / 2.0
+
+            dist_reward = A_distance_triu.sum((-1, -2)) / denominator
+            # >>> n_fantasy_at_design_pts x batch_size
+
+        q_hes = Y + self.dist_weight * dist_reward
+        # >>> n_fantasy_at_design_pts x batch_size
+
+        return q_hes
+
+
+class qLossFunctionMinMax(nn.Module):
+    r"""Batch loss function for the task of finding min and max
+    relative to the values of the objective function."""
+
+    def __init__(self) -> None:
+        r"""Loss function for task of finding min and max
+        relative to the values of the objective function."""
+
+        super().__init__()
+
+    def forward(self, A: Tensor, Y: Tensor) -> Tensor:
+        r"""Evaluate batch loss function on a tensor of actions.
+
+        Args:
+            A: Actor tensor with shape `n_fantasy_at_design_pts x batch_size
+                x num_actions x action_dim`.
+            Y: Fantasized sample with shape `n_fantasy_at_action_pts x
+                n_fantasy_at_design_pts x batch_size x num_actions`.
+
+        Returns:
+            A Tensor of shape `n_fantasy_at_action_pts x batch`.
+        """
+
+        if A.shape[-2] != 2:
+            raise RuntimeError("qLossFunctionMinMax only supports 2 actions.")
+
+        q_hes = (Y[..., 1:].sum(dim=-1) - Y[..., 0]).mean(dim=0)
+        # >>> n_fantasy_at_design_pts x batch_size
+
+        return q_hes
```

Comment:
```suggestion
        return q_hes
        
```

A newline at the end of the file is required by black. (You should also be able to run `ufmt format .` to fix this.)